src/main/scala/firrtl_interpreter/CircuitState.scala

```diff
@@ -116,19 +123,25 @@ case class CircuitState(
     * cycle all memories
     */
   def cycle(): Unit = {
+    vcdLoggerOption.foreach { vcd =>
+      vcd.incrementTime(10)
+      vcd.raiseClock
+    }
     registers.keys.foreach { key =>
       registers(key) = nextRegisters(key)
     }
 //    nextRegisters.clear()
 //    ephemera.clear()
     cycleMemories()
-    vcdLoggerOption.foreach { vcd =>
-      vcd.incrementTime()
-    }
     nameToConcreteValue = mutable.HashMap((inputPorts ++ outputPorts ++ registers).toSeq:_*)
     isStale = true
 
     stateCounter += 1
+
+    vcdLoggerOption.foreach { vcd =>
+      vcd.incrementTime(10)
```

Comment:
yw;)
actually I'd made kinda similar patch yesterday, including added hierarchy to 'Scope's, but was facing some problems with register updates. I see a similar issue in this PR as well. I just tried the VendingMachine example and the "state" register appears to be showing nextstate istead of it's current state in the vcd. One more thing, the first timestamp in the vcd is -1 which my vcd viewer doesn't seem to like. Not quite sure what the vcd spec says about this.